quattor/functions/network.pan

```diff
@@ -1,206 +1,123 @@
-################################################################################
+@documentation{
+    descr = returns the first (alphabetically sorted) NIC interface found with boot=true
+}
 function boot_nic = {
-    base = "/hardware/cards/nic";
-    nics = value(base);
-    device = "";
-    ok = first(nics,i,v);
-    while (ok) {
-        path = base+"/"+to_string(i)+"/boot";
-        if (exists(to_string(path))) {
-            if (value(path)) {
-                device = i;
-            };
+    foreach (ifname; data; value("/hardware/cards/nic")) {
+        if(exists(data['boot']) && data['boot']) {
+            return(ifname);
         };
-        ok = next(nics,i,v);
     };
-    if (device=="") {
-        error("boot flag not set for nic");
-    };
-    device;
+    error("boot flag not set for any nic");
 };
 
 
-################################################################################
-# get_subnet_params(subnet_list:list, ipaddr:string)
-#
-# Checks which subnet ipaddr is a member of and returns the corresponding subnet
-# parameters for netmask, gateway...
-#
-# subnet_list must contain a list of nlist. Each list element describes a subnet.
-# Possible keys are all the properties supported by structure_interface plus
-# 'subnet' that must contain a regexp matching the subnet addresses.
-# This is a 'dirty solution' to work around the lack of bitwise AND in pan...
-################################################################################
-function get_subnet_params = {
-    if ( ARGC != 2 ) {
-        error("get_subnet_params requires 2 arguments");
-    };
-
-    subnet_list = ARGV[0];
-    ipaddr = ARGV[1];
+@documentation{
+    desc = dict defining a non default MTU size for each interface in the system.\
+           dict keys can be an interface name, an interface type (e.g. eth, em), DEFAULT or BOOT.\
+           BOOT entry is applied to main interface (no explicit value must exist for it).\
+           DEFAULT entry is applied to all interfaces without an explicit value defined.
+    values = MTU size (integer) for the interface or undef to prevent explicit definition of the MTU size.
+    default = none
+    required = no
+}
+variable MTU ?= dict();
+
+@documentation{
+    descr = This function configures all the network interfaces declared in\
+            /hardware/cards/nic. Parameters are taken from variable\
+            NETWORK_PARAMS (dict) for the main (boot) interface, others\
+            are configured with dhcp. For every interface, if there is\
+            an entry in variable MTU, it is also applied to the interface.\
+            Modifies and returns SELF.
+}
+function copy_network_params = {
+    nics = value('/hardware/cards/nic');
 
-    if ( !is_list(subnet_list) ) {
-        error("get_subnet_params first argument must be a list");
+    if (!is_defined(nics)) {
+        error("No network cards defined in the configuration");
+    };
+    if (!is_nlist(nics)) {
```

Comment:
It's confusing to refer to this as a dict everywhere and then use is_nlist() in preference to is_dict.